### PR TITLE
[AMORO-2017][Spark] Format code in the spark module with the spotless plugin

### LINKED ISSUE #2017: [Subtask]: Apply spotless to Spark Modules
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/amoro/issues?q=is%3Aissue) and found no similar issues.

### Description

Apply spotless to spark module and config mvn validate as spotless.

### Parent issue

#1402

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Close #2017 
enable plugin spotless and skip checkstyle for those module
- `mvn spotless:apply -pl spark/common`
- `mvn spotless:apply -pl spark/v3.1/spark`
- `mvn spotless:apply -pl spark/v3.2/spark`
- `mvn spotless:apply -pl spark/v3.3/spark`

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by XBaith** [comment]
cc @XBaith @zhoujinsong

**Review by baiyangtx — CHANGES_REQUESTED** [review]
The purpose of this issue is to gradually replace the maven-checkstyle-plugin with the spotless plugin. Therefore, you need to enable spotless in the plugins section of the Spark module and remove the maven-checkstyle-plugin.

Those modules should be changed.

- spark-common
- spark/v3.1/spark
- spark/v3.2/spark
- spark/v3.3/spark

File: spark/common/pom.xml [hunk 1]
```diff
@@ -253,4 +253,13 @@
...
     </profiles>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.diffplug.spotless</groupId>
+                <artifactId>spotless-maven-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
```
[the remark this answers is not in the document]
Comment (XBaith) [reply spark/common/pom.xml]: Sure

File: spark/v3.2/spark/pom.xml [hunk 2]
```diff
@@ -419,11 +419,8 @@
...
             </plugin>
             <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-checkstyle-plugin</artifactId>
-                <configuration>
-                    <includeTestSourceDirectory>true</includeTestSourceDirectory>
-                </configuration>
+                <groupId>com.diffplug.spotless</groupId>
+                <artifactId>spotless-maven-plugin</artifactId>
             </plugin>
```
[the remark this answers is not in the document]
Comment (XBaith) [reply spark/v3.2/spark/pom.xml]: Good catch

File: spark/common/pom.xml [hunk 3]
```diff
@@ -253,4 +253,20 @@
             </build>
         </profile>
     </profiles>
+
+    <build>
```
Comment (zhoujinsong) [inline spark/common/pom.xml:257]: We can put this configuration section in `spark/pom.xml` to avoid configuring each submodule separately.

Comment (XBaith) [reply spark/common/pom.xml]: Thanks for your suggestion

**Review by baiyangtx — APPROVED** [review]
LGTM, thanks for your contributions.

**Pull request merged**